lib/facil/http/http1.c

```diff
@@ -342,6 +345,22 @@ static int http1_http2websocket_server(http_s *h, websocket_settings_s *args) {
     goto bad_request;
   stmp = fiobj_obj2cstr(tmp);
 
+  FIOBJ tmp_protocol = fiobj_hash_get2(h->headers, sec_protocol);
+  if (tmp_protocol) {
+    char *protocol_str = fiobj_obj2cstr(tmp_protocol).data;
+
+    if (strlen(protocol_str)) {
```

Comment:
Avoid `strlen` for strings with a known length. Length data is returned as part of the `fiobj_obj2cstr` return value and there's no need to re-compute the string length (it is cached).